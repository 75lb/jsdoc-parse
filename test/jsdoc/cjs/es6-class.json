[
  {
    "comment": "/**\n * @module es6-class\n */",
    "meta": {
      "range": [
        0,
        28
      ],
      "filename": "es6-class.js",
      "lineno": 1,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {}
    },
    "kind": "module",
    "name": "es6-class",
    "longname": "module:es6-class"
  },
  {
    "comment": "/**\n * a global class\n * @alias module:es6-class\n */",
    "meta": {
      "range": [
        83,
        442
      ],
      "filename": "es6-class.js",
      "lineno": 9,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000001",
        "name": "GlobalClass",
        "type": "ClassDeclaration",
        "paramnames": [
          "one",
          "two"
        ]
      }
    },
    "classdesc": "a global class",
    "alias": "module:es6-class",
    "name": "module:es6-class",
    "longname": "module:es6-class",
    "kind": "class",
    "description": "the global class constructor",
    "params": [
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "first param",
        "name": "one"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "second param",
        "name": "two"
      }
    ]
  },
  {
    "comment": "/**\n   * the global class constructor\n   * @param {number} - first param\n   * @param {string} - second param\n   */",
    "meta": {
      "range": [
        222,
        303
      ],
      "filename": "es6-class.js",
      "lineno": 15,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000004",
        "name": "GlobalClass",
        "type": "MethodDefinition",
        "paramnames": [
          "one",
          "two"
        ]
      },
      "vars": {
        "": null
      }
    },
    "description": "the global class constructor",
    "params": [
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "first param",
        "name": "one"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "second param",
        "name": "two"
      }
    ],
    "name": "GlobalClass",
    "longname": "module:es6-class.GlobalClass",
    "kind": "class",
    "memberof": "module:es6-class",
    "scope": "static",
    "undocumented": true
  },
  {
    "comment": "/** an instance property */",
    "meta": {
      "range": [
        283,
        299
      ],
      "filename": "es6-class.js",
      "lineno": 17,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000011",
        "name": "this.propOne",
        "type": "Literal",
        "value": 1,
        "paramnames": []
      }
    },
    "description": "an instance property",
    "name": "propOne",
    "longname": "module:es6-class.GlobalClass#propOne",
    "kind": "member",
    "memberof": "module:es6-class.GlobalClass",
    "scope": "instance"
  },
  {
    "comment": "/** a static method */",
    "meta": {
      "range": [
        332,
        349
      ],
      "filename": "es6-class.js",
      "lineno": 21,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000016",
        "name": "GlobalClass.shat",
        "type": "MethodDefinition",
        "paramnames": []
      },
      "vars": {
        "": null
      }
    },
    "description": "a static method",
    "name": "shat",
    "longname": "module:es6-class.shat",
    "kind": "function",
    "memberof": "module:es6-class",
    "scope": "static"
  },
  {
    "comment": "/** parent method one */",
    "meta": {
      "range": [
        380,
        395
      ],
      "filename": "es6-class.js",
      "lineno": 24,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000020",
        "name": "GlobalClass#methodOne",
        "type": "MethodDefinition",
        "paramnames": []
      },
      "vars": {
        "": null
      }
    },
    "description": "parent method one",
    "name": "methodOne",
    "longname": "module:es6-class#methodOne",
    "kind": "function",
    "memberof": "module:es6-class",
    "scope": "instance"
  },
  {
    "comment": "/** parent method two */",
    "meta": {
      "range": [
        425,
        440
      ],
      "filename": "es6-class.js",
      "lineno": 26,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000024",
        "name": "GlobalClass#methodTwo",
        "type": "MethodDefinition",
        "paramnames": []
      },
      "vars": {
        "": null
      }
    },
    "description": "parent method two",
    "name": "methodTwo",
    "longname": "module:es6-class#methodTwo",
    "kind": "function",
    "memberof": "module:es6-class",
    "scope": "instance"
  },
  {
    "comment": "/**\n * the child of global class\n * @extends GlobalClass\n */",
    "meta": {
      "range": [
        505,
        684
      ],
      "filename": "es6-class.js",
      "lineno": 33,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000028",
        "name": "GlobalChildClass",
        "type": "ClassDeclaration",
        "paramnames": []
      }
    },
    "classdesc": "the child of global class",
    "augments": [
      "GlobalClass"
    ],
    "name": "GlobalChildClass",
    "longname": "module:es6-class~GlobalChildClass",
    "kind": "class",
    "scope": "inner",
    "memberof": "module:es6-class"
  },
  {
    "comment": "",
    "meta": {
      "range": [
        552,
        627
      ],
      "filename": "es6-class.js",
      "lineno": 34,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000032",
        "name": "GlobalChildClass",
        "type": "MethodDefinition",
        "paramnames": []
      },
      "vars": {
        "": null
      }
    },
    "undocumented": true,
    "name": "GlobalChildClass",
    "longname": "module:es6-class~GlobalChildClass.GlobalChildClass",
    "kind": "class",
    "memberof": "module:es6-class~GlobalChildClass",
    "scope": "static"
  },
  {
    "comment": "/** an instance property */",
    "meta": {
      "range": [
        605,
        623
      ],
      "filename": "es6-class.js",
      "lineno": 36,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000037",
        "name": "this.propThree",
        "type": "Literal",
        "value": 1,
        "paramnames": []
      }
    },
    "description": "an instance property",
    "name": "propThree",
    "longname": "module:es6-class~GlobalChildClass.GlobalChildClass#propThree",
    "kind": "member",
    "memberof": "module:es6-class~GlobalChildClass.GlobalChildClass",
    "scope": "instance"
  },
  {
    "comment": "/** overridden child methodTwo */",
    "meta": {
      "range": [
        667,
        682
      ],
      "filename": "es6-class.js",
      "lineno": 40,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000042",
        "name": "GlobalChildClass#methodTwo",
        "type": "MethodDefinition",
        "paramnames": []
      },
      "vars": {
        "": null
      }
    },
    "description": "overridden child methodTwo",
    "name": "methodTwo",
    "longname": "module:es6-class~GlobalChildClass#methodTwo",
    "kind": "function",
    "memberof": "module:es6-class~GlobalChildClass",
    "scope": "instance"
  },
  {
    "comment": "",
    "meta": {
      "range": [
        686,
        714
      ],
      "filename": "es6-class.js",
      "lineno": 43,
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs",
      "code": {
        "id": "astnode100000047",
        "name": "module.exports",
        "type": "Identifier",
        "value": "GlobalClass",
        "paramnames": []
      }
    },
    "undocumented": true,
    "name": "module:es6-class",
    "longname": "module:es6-class",
    "kind": "member"
  },
  {
    "kind": "package",
    "longname": "package:undefined",
    "files": [
      "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs/es6-class.js"
    ]
  }
]
