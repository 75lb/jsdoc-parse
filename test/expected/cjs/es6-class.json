[
  {
    "id": "module:es6-class",
    "longname": "module:es6-class",
    "name": "es6-class",
    "kind": "module",
    "meta": {
      "lineno": 1,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 0
  },
  {
    "id": "module:es6-class--GlobalClass",
    "longname": "module:es6-class",
    "name": "GlobalClass",
    "kind": "class",
    "isExported": true,
    "description": "a global class",
    "memberof": "module:es6-class",
    "alias": "module:es6-class",
    "meta": {
      "lineno": 9,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 1
  },
  {
    "id": "module:es6-class--GlobalClass.GlobalClass",
    "longname": "module:es6-class.GlobalClass",
    "name": "GlobalClass",
    "kind": "constructor",
    "description": "the global class constructor",
    "memberof": "module:es6-class--GlobalClass",
    "params": [
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "first param",
        "name": "one"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "second param",
        "name": "two"
      }
    ],
    "meta": {
      "lineno": 15,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 2
  },
  {
    "id": "module:es6-class--GlobalClass.GlobalClass#propOne",
    "longname": "module:es6-class.GlobalClass#propOne",
    "name": "propOne",
    "scope": "instance",
    "kind": "member",
    "description": "an instance property",
    "memberof": "module:es6-class--GlobalClass",
    "meta": {
      "lineno": 17,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 3
  },
  {
    "id": "module:es6-class--GlobalClass~GlobalChildClass.GlobalChildClass#propThree",
    "longname": "module:es6-class~GlobalChildClass.GlobalChildClass#propThree",
    "name": "propThree",
    "scope": "instance",
    "kind": "member",
    "description": "an instance property",
    "memberof": "module:es6-class--GlobalClass~GlobalChildClass",
    "meta": {
      "lineno": 36,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 8
  },
  {
    "id": "module:es6-class--GlobalClass#methodOne",
    "longname": "module:es6-class#methodOne",
    "name": "methodOne",
    "scope": "instance",
    "kind": "function",
    "description": "parent method one",
    "memberof": "module:es6-class--GlobalClass",
    "meta": {
      "lineno": 24,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 5
  },
  {
    "id": "module:es6-class--GlobalClass#methodTwo",
    "longname": "module:es6-class#methodTwo",
    "name": "methodTwo",
    "scope": "instance",
    "kind": "function",
    "description": "parent method two",
    "memberof": "module:es6-class--GlobalClass",
    "meta": {
      "lineno": 26,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 6
  },
  {
    "id": "module:es6-class--GlobalClass~GlobalChildClass#methodTwo",
    "longname": "module:es6-class~GlobalChildClass#methodTwo",
    "name": "methodTwo",
    "scope": "instance",
    "kind": "function",
    "description": "overridden child methodTwo",
    "memberof": "module:es6-class--GlobalClass~GlobalChildClass",
    "meta": {
      "lineno": 40,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 9
  },
  {
    "id": "module:es6-class--GlobalClass.shat",
    "longname": "module:es6-class.shat",
    "name": "shat",
    "scope": "static",
    "kind": "function",
    "description": "a static method",
    "memberof": "module:es6-class--GlobalClass",
    "meta": {
      "lineno": 21,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 4
  },
  {
    "id": "module:es6-class--GlobalClass~GlobalChildClass",
    "longname": "module:es6-class~GlobalChildClass",
    "name": "GlobalChildClass",
    "scope": "inner",
    "kind": "class",
    "augments": [
      "GlobalClass"
    ],
    "description": "the child of global class",
    "memberof": "module:es6-class--GlobalClass",
    "meta": {
      "lineno": 33,
      "filename": "es6-class.js",
      "path": "/Users/lloyd/Documents/75lb/jsdoc-parse/test/fixture/cjs"
    },
    "order": 7
  }
]
